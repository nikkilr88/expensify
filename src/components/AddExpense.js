import React, { Component } from 'react'

class AddExpense extends Component {
  render() {
    return <p>Add</p>
  }
}

export default AddExpense
