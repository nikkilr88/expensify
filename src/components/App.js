import React, { Component } from 'react'

class App extends Component {
  render() {
    return (
      <p>
        Lorem ipsum dolor sit amet consectetur adipisicing elit. Quo, error?
        Ducimus quaerat saepe quisquam eveniet eius consequuntur. Voluptatibus
        delectus sint voluptas quibusdam in quae natus sed! Sunt facilis
        repudiandae quaerat?
      </p>
    )
  }
}

export default App
