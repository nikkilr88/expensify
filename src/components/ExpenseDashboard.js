import { Link } from 'react-router-dom'

const ExpenseDashboard = () => <p>this is the dashboard</p>

export default ExpenseDashboard
