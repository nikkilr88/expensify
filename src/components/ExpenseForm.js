import React, { Component } from 'react'

class ExpenseForm extends Component {
  state = {
    description: '',
    amount: '',
    note: ''
  }

  onChange = e => {
    e.persist()
    this.setState(() => ({
      [e.target.name]: e.target.value
    }))
  }

  onAmountChange = e => {
    const amount = e.target.value
    if (amount.match(/^\d*(\.\d{0,2})?$/)) {
      this.setState(() => ({
        amount
      }))
    }
  }

  render() {
    return (
      <div>
        <form>
          <input
            name='description'
            type='text'
            placeholder='Description'
            value={this.state.description}
            onChange={this.onChange}
            autoFocus
          />
          <input
            name='amount'
            type='text'
            value={this.state.amount}
            onChange={this.onAmountChange}
            placeholder='Amount'
          />
          <textarea
            name='note'
            value={this.state.note}
            onChange={this.onChange}
            placeholder='Add a note for you expense (optional)'
          />

          <button>Save Expense</button>
        </form>
      </div>
    )
  }
}

export default ExpenseForm
