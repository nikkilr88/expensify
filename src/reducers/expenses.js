const expensesState = []

const expensesReducer = (state = expensesState, action) => {
  switch (action.type) {
    case 'ADD_EXPENSE':
      return [...state, action.expense]
    case 'REMOVE_EXPENSE':
      return state.filter(exp => exp.id !== action.id)
    case 'EDIT_EXPENSE':
      return state.map(expense => {
        if (expense.id === action.id) {
          return {
            ...expense,
            ...action.updates
          }
        }
      })
    default:
      return state
  }
}

export default expensesReducer
