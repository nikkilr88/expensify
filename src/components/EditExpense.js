import React, { Component } from 'react'
import { connect } from 'react-redux'
import ExpenseForm from './ExpenseForm'
import { editExpense, removeExpense } from '../actions/expenses'

const EditExpense = props => {
  return (
    <div>
      <p>Edit expense {props.match.params.id}</p>

      <ExpenseForm
        expense={props.expense}
        onSubmit={expense => {
          props.dispatch(editExpense(props.expense.id, expense))
          props.history.push('/dashboard')
        }}
      />

      <button
        onClick={() => {
          props.dispatch(removeExpense({ id: props.expense.id }))
          props.history.push('/dashboard')
        }}
      >
        Remove
      </button>
    </div>
  )
}

const mapStateToProps = (state, props) => ({
  expense: state.expenses.find(expense => expense.id === props.match.params.id)
})

export default connect(mapStateToProps)(EditExpense)
