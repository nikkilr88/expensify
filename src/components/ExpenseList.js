import React from 'react'
import { connect } from 'react-redux'
import ExpenseListItem from './ExpenseListItem'
import { getVisibleExpenses } from '../selectors/expenses'

export const ExpenseList = ({ expenses }) => {
  return (
    <div>
      {expenses.length === 0
        ? ''
        : expenses.map((expense, i) => (
            <ExpenseListItem key={expense.id} {...expense} />
          ))}
    </div>
  )
}

const mapStateToProps = state => {
  return {
    expenses: getVisibleExpenses(state.expenses, state.filters)
  }
}

export default connect(mapStateToProps)(ExpenseList)
