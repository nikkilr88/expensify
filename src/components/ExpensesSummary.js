import React from 'react'
import numeral from 'numeral'
import { connect } from 'react-redux'
import { getVisibleExpenses, getExpensesTotal } from '../selectors/expenses'

export const ExpensesSummary = ({ expenseCount, expenseTotal }) => {
  return (
    <h3>
      Viewing {expenseCount} expense(s) totaling{' '}
      {numeral(expenseTotal / 100).format('$0,0.00')}
    </h3>
  )
}

const mapStateToProps = state => {
  const expenses = getVisibleExpenses(state.expenses, state.filters)
  return {
    expenseCount: expenses.length,
    expenseTotal: getExpensesTotal(expenses)
  }
}

export default connect(mapStateToProps)(ExpensesSummary)
