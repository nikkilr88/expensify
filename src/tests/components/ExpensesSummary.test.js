import React from 'react'
import { shallow } from 'enzyme'
import { ExpensesSummary } from '../../components/ExpensesSummary'
import expenses from '../fixtures/expenses'

test('should render ExpensesSummary correctly with no expenses', () => {
  const wrapper = shallow(<ExpensesSummary />)
  expect(wrapper).toMatchSnapshot()
})

test('should render ExpensesSummary correctly with expenses', () => {
  const wrapper = shallow(
    <ExpensesSummary expenseCount={expenses.length} expenseTotal={100} />
  )
  expect(wrapper).toMatchSnapshot()
})
