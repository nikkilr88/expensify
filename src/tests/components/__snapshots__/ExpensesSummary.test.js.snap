// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should render ExpensesSummary correctly with expenses 1`] = `
<h3>
  Viewing 
  3
   expense(s) totaling
   
  $1.00
</h3>
`;

exports[`should render ExpensesSummary correctly with no expenses 1`] = `
<h3>
  Viewing 
   expense(s) totaling
   
  $0.00
</h3>
`;
